source/docs/software/advanced-controls/state-space/state-space-flywheel-walkthrough.rst

```diff
@@ -0,0 +1,204 @@
+
+Modeling with System identification
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To rewrite this in state-space notation using :term:`system identification`, we recall from the flywheel :ref:`state-space notation example <docs/software/advanced-controls/state-space/state-space-intro:State-space notation example -- Flywheel from kV and kA>`, where we rewrote the equation :math:`V = kV \cdot v + kA \cdot a` in terms of :math:`a` as :math:`\mathbf{a} = \mathbf{\dot{v}} = \begin{bmatrix}\frac{-kV}{kA}\end{bmatrix} v + \begin{bmatrix}\frac{1}{kA}\end{bmatrix} V`, where :math:`v` is flywheel velocity, :math:`\dot{v}` and :math:`a` are acceleration, and :math:`V` is voltage. Rewriting this with the standard convention of :math:`x` for the state vector and :math:`u` for inputs, we find:
+
+.. math::
+    \mathbf{\dot{x}} = \begin{bmatrix}\frac{-kV}{kA} \end{bmatrix} \mathbf{x} + \begin{bmatrix}\frac{1}{kA} \end{bmatrix} \mathbf{u}
+
+The second part of state-space notation relates the system's current :term:`state` and :term:`inputs <input>` to the :term:`output`. In the case of a flywheel, our output (or thing that our sensors can measure) is flywheel velocity, so :math:`\begin{bmatrix}C \end{bmatrix}` is just :math:`\begin{bmatrix}1 \end{bmatrix}`. The system feedthrough matrix :math:`\mathbf{D}`, which relates input to output, is often simply :math:`\begin{bmatrix}0 \end{bmatrix}` (except in exceptional cases).
+
+.. math::
+    \mathbf{\dot{x}} &= \begin{bmatrix}\frac{-kV}{kA} \end{bmatrix} \mathbf{x} + \begin{bmatrix}\frac{1}{kA} \end{bmatrix} \mathbf{u} \\
+    \mathbf{y} &= \begin{bmatrix}1\end{bmatrix} \mathbf{x} + \begin{bmatrix}0\end{bmatrix} \mathbf{u}
+
+Where :math:`\mathbf{y}` is the flywheel's velocity, as measured by a sensor of some kind.
+
+The ``LinearSystem`` class contains methods for easily creating state-space systems identified using :term:`system identification`. This example shows a flywheel model with a kV of 1 and a kA of 0.5:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/40eae3ab109b8ebf3010b7cd29a8b4d7fde0a050/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/statespaceflywheelsysid/Robot.java
+         :language: java
+         :lines: 36-47
+         :linenos:
+         :lineno-start: 36
+
+   .. group-tab:: C++
+
+      .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/40eae3ab109b8ebf3010b7cd29a8b4d7fde0a050/wpilibcExamples/src/main/cpp/examples/StateSpaceFlywheelSysId/cpp/Robot.cpp
+         :language: cpp
+         :lines: 36-42
+         :linenos:
+         :lineno-start: 36
+
```

Comment:
Perhaps add a note with the relevant `#include` headers for C++. VS Code will automatically import these for Java but C++ users don't have this luxury.